File: app/src/main/java/ma/ensa/e_gestionnairemedec/classe/Rappel.kt
package ma.ensa.e_gestionnairemedec.classe

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class Rappel(
    private var id: Int,
    private var utilisateurId: Int,
    private var titre: String,
    private var description: String,
    private var dateHeure: Date,
    private var etat: String,
    private var createdAt: Date,
    private var medicament: String
) {
    // Constructeur secondaire pour créer un rappel avec des valeurs par défaut
    constructor() : this(
        id = 0,
        utilisateurId = 0,
        titre = "",
        description = "",
        dateHeure = Date(),
        etat = "en_attente",
        createdAt = Date(),
        medicament = ""
    )

    // Getters
    fun getId() = id
    fun getUtilisateurId() = utilisateurId
    fun getTitre() = titre
    fun getDescription() = description
    fun getEtat() = etat
    fun getMedicament() = medicament

    fun getDateHeure(): Date = dateHeure

    // Retourner la date de création sous forme de chaîne
    fun getCreatedAt(): String {
        val dateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault())
        return dateFormat.format(createdAt)
    }

    // Formater `dateHeure` sous forme de chaîne pour l'affichage
    fun getFormattedDateHeure(): String {
        val dateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault())
        return dateFormat.format(dateHeure)
    }

    // Setters
    fun setId(id: Int) {
        this.id = id
    }

    fun setUtilisateurId(utilisateurId: Int) {
        this.utilisateurId = utilisateurId
    }

    fun setTitre(titre: String) {
        this.titre = titre
    }

    fun setDescription(description: String) {
        this.description = description
    }

    fun setDateHeure(dateHeure: Date) {
        this.dateHeure = dateHeure
    }

    fun setEtat(etat: String) {
        this.etat = etat
    }

    fun setCreatedAt(createdAt: Date) {
        this.createdAt = createdAt
    }

    fun setMedicament(medicament: String) {
        this.medicament = medicament
    }
}

File: Backend/classes/Rappel.php
<?php
class Rappel {
    private $id;
    private $utilisateur_id;
    private $titre;
    private $description;
    private $date_heure;
    private $etat;
    private $created_at;
    private $medicament;

    // Constructeur avec des paramètres optionnels pour certains attributs
    public function __construct($id = null, $utilisateur_id, $titre, $description, $date_heure, $etat, $created_at, $medicament) {
        $this->id = $id;
        $this->utilisateur_id = $utilisateur_id;
        $this->titre = $titre;
        $this->description = $description;
        $this->date_heure = $date_heure;
        $this->etat = $etat;
        $this->created_at = $created_at;
        $this->medicament = $medicament;
    }

    // Getters et Setters pour chaque propriété
    public function getId() {
        return $this->id;
    }

    public function setId($id) {
        $this->id = $id;
    }

    public function getUtilisateurId() {
        return $this->utilisateur_id;
    }

    public function setUtilisateurId($utilisateur_id) {
        $this->utilisateur_id = $utilisateur_id;
    }

    public function getTitre() {
        return $this->titre;
    }

    public function setTitre($titre) {
        $this->titre = $titre;
    }

    public function getDescription() {
        return $this->description;
    }

    public function setDescription($description) {
        $this->description = $description;
    }

    public function getDateHeure() {
        return $this->date_heure;
    }

    public function setDateHeure($date_heure) {
        $this->date_heure = $date_heure;
    }

    public function getEtat() {
        return $this->etat;
    }

    public function setEtat($etat) {
        $this->etat = $etat;
    }

    public function getCreatedAt() {
        return $this->created_at;
    }

    public function setCreatedAt($created_at) {
        $this->created_at = $created_at;
    }

    public function getMedicament() {
        return $this->medicament;
    }

    public function setMedicament($medicament) {
        $this->medicament = $medicament;
    }
}
?>
